packages/fuel-indexer/src/service.rs

```diff
@@ -277,6 +277,80 @@ async fn create_service_task(
+                            killer.store(true, Ordering::SeqCst);
+                        } else {
+                            warn!(
+                                "Revert Indexer: No indexer with the name Index({uid})"
+                            );
+                        }
+
+                        let mut conn = pool
+                            .acquire()
+                            .await
+                            .expect("Failed to acquire connection from pool");
+
+                        match queries::penultimate_index_id_for(
+                            &mut conn,
+                            &request.namespace,
+                            &request.identifier,
+                        )
+                        .await
+                        {
+                            Ok(id) => {
+                                let assets =
+                                    queries::latest_assets_for_index(&mut conn, &id)
+                                        .await
+                                        .expect("Could not get latest assets for index");
+                                let manifest: Manifest =
+                                    serde_yaml::from_slice(&assets.manifest.bytes)
+                                        .expect("Failed to deserialize manifest");
+
+                                let (handle, _module_bytes, killer) = WasmIndexExecutor::create(
+                                    &config.fuel_node,
+                                    &config.database.to_string(),
+                                    &manifest,
+                                    ExecutorSource::Registry(assets.wasm.bytes),
+                                    config.stop_idle_indexers,
+                                )
+                                .await
+                                .expect(
+                                    "Failed to spawn executor from index asset registry",
+                                );
+
+                                futs.push(handle);
+                                killers.insert(manifest.uid(), killer);
+
+                                match queries::remove_index(
+                                    &mut conn,
+                                    &request.namespace,
+                                    &request.identifier,
+                                )
+                                .await
+                                {
+                                    Ok(_) => info!(
+                                        "Removed index: {}.{}",
+                                        &request.namespace, &request.identifier
+                                    ),
+                                    Err(e) => error!(
+                                        "Failed to remove index: {}.{}: {}",
+                                        &request.namespace, &request.identifier, e
+                                    ),
+                                }
```

Comment:
So `remove_index` removes the _entire_ index, as it deletes any rows that match to the `index_id` for the namespace and identifier. In contrast, I'd expect a removal of the reverted index to remove the most recent `version` of the WASM asset for said `index_id`. My apologies if I caused a misunderstanding when suggesting it earlier.

This would require a separate function to be created or for `remove_index` to take at least another argument to instruct what should be deleted.

@ra0x3: Thoughts?